pkg/cloudprovider/provider/vsphere/provider.go

```diff
@@ -398,6 +405,24 @@ func (p *provider) Create(machine *v1alpha1.Machine, _ *cloud.MachineCreateDelet
 		return nil, fmt.Errorf("error when waiting for vm powerOn task: %v", err)
 	}
 
+	// Add a custom field to indicate to our Get that creation succeeded
+	// If the field is not set, Get will delete the instance
```

Comment:
The creation can fail at various stages and we may not have a reference to the cloned VM even if it gets created, e.G. when the clone itself errord or timed out. This is the best I could think of to make sure either everything suceeds or we retry. LMK what you think